### PR TITLE
Resource view GUI

### PR DESCRIPTION
**Pull request by Compro-Prasad**

### CODE REVIEW

File: src/resources/resources-view.glade [hunk 1]
```diff
@@ -1,26 +1,174 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<!-- Generated with glade 3.20.0 -->
+<!-- Generated with glade 3.22.1 -->
```
Comment (joesteeve) [inline src/resources/resources-view.glade:2]: Oh fish. Debian stretch is still on 3.20 .. What distro/version are you on? We need to freeze on a glade version. Probably just build-install a specific version.

File: src/resources-view.h [hunk 2]
```diff
@@ -21,6 +21,19 @@ class ResourcesViewCallbacks
 public:
   virtual ~ResourcesViewCallbacks () {}
 
+  virtual void view_node_selected (
+                                   std::vector<WBS::Path> && path_list) = 0;
+  virtual void view_node_activated (
+                                    const WBS::Path & path) = 0;
```
Comment (joesteeve) [inline src/resources-view.h]: Indenting.

File: src/resources-view.h [hunk 3]
```diff
@@ -31,6 +44,18 @@ class ResourcesView
...
   virtual void set_handler (ResourcesViewCallbacks* handler) = 0;
+  
+  virtual void enable_add_resource (bool enable) = 0;
+  virtual void enable_add_resource_group (bool enable) = 0;
+  virtual void enable_up (bool enable) = 0;
+  virtual void enable_down (bool enable) = 0;
+  virtual void enable_cut (bool enable) = 0;
+  virtual void enable_copy (bool enable) = 0;
+  virtual void enable_paste (bool enable) = 0;
+  virtual void enable_delete (bool enable) = 0;
```
Comment (joesteeve) [inline src/resources-view.h:48]: Make the method names specific. i..e enable_add_resource_button. Verbose names are okay. Easier to read later on.

File: src/resources-view.h [hunk 4]
```diff
@@ -31,6 +44,18 @@ class ResourcesView
...
+
+  virtual void add_resource (const WBS::Path & path) = 0;
+  virtual void add_resource_group (const WBS::Path & t_path) = 0;
 };
```
Comment (joesteeve) [inline src/resources-view.h:53]: path, t_path ? Stay consistent.

File: src/resources-controller_impl.h [hunk 5]
```diff
@@ -35,8 +35,8 @@ class ResourcesControllerImpl
     return m_view->widget ();
   }
 
-  void view_node_selected (std::vector<WBS::Path> && path_list);
-  void view_node_activated (const WBS::Path & path);
+  void view_node_selected ();
+  void view_node_activated ();
```
Comment (joesteeve) [inline src/resources-controller_impl.h:39]: If there are no parameters, explicitly say 'void'. Dont leave it empty.

**Comment by Compro-Prasad** [comment]
Merge this PR. The resource view is complete. The controller is now as a new PR #33 . Updates will be made there.

**Pull request merged**